app/controllers/offers_controller.rb

```diff
@@ -0,0 +1,82 @@
+class OffersController < ApplicationController
+  protect_from_forgery with: :null_session
+
+  def index
+    render json: get_all_offers
+  end
+
+  def show
+    offer = Offer.find(params[:id])
+    render json: format_offer(offer)
+  end
+
+  def show_by_instructor
+    offers = []
+    get_all_offers.each do |offer|
+      if taught_by(offer, params[:instructor_id].to_i)
```

Comment:
You could use `get_all_offers.filter { |o| taught_by(offer, params[:instructor_id].to_i) }` to obtain an array of offers matching a condition